test/kube_testing/k8s.go

```diff
@@ -283,8 +283,12 @@ func NewK8sWithoutRoles(prepare bool) (*K8s, error) {
 		start := time.Now()
 		client.Prepare("nsmgr", "nsmd", "vppagent", "vpn", "icmp", "nsc", "source", "dest")
 		client.CleanupCRDs()
+		client.CleanupServices("nsm-admission-webhook-svc")
+		client.CleanupDeployments()
+		client.CleanupMutatingWebhookConfigurations()
+		client.CleanupSecrets("nsm-admission-webhook-certs")
```

Comment:
Hmm, in general, that is right. On the other hand aren't we adding too many corner cases here?
For example, the admission webhook is used only in 2 tests. Can we just have a `defer`  to clean up those 2 specific tests and not put the service removal code here for each and every test we run?

Reply:
There is already used `defer` to clean up all these resources.
The described scenario for the issue means that there is some reason for `defer` not being executed - it can be either a hung test or a test killed by the signal.

Reply:
I'm ok if this not involves more time to cleanup.

Reply:
It seems to increase cleanup time for ~0.1-0.5 second for test in different clouds.
Added a bool param to `NewK8s()` to enable / disable "extra cleanup".